### PR TITLE
docs(bigtable): add page describing retry policies

### LINKED ISSUE #11265: Document bigtable retry and backoff policies

### PR DESCRIPTION
**Pull request by coryan**
Fixes #11265 Part of the work for #10157 

----
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/11955)

### CODE REVIEW

File: google/cloud/bigtable/doc/override-retry-policies.dox [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+The library uses three different options to control the retry loop. The options
+have per-client names
+
+@section bigtable-override-retry-retry-policy Configuring the transient errors and retry duration
+
+The `*RetryPolicyOption` controls:
+
+- Which errors are to be treated as transient errors.
+- How long will the library will keep retrying transient errors.
```
Comment (dbolduc) [inline google/cloud/bigtable/doc/override-retry-policies.dox:36]: just noticed this.... s/will the library will/the library will/

File: google/cloud/bigtable/doc/override-retry-policies.dox [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+@see google::cloud::bigtable::DataRetryPolicyOption
+@see google::cloud::bigtable::DataRetryPolicy
+@see google::cloud::bigtable::DataLimitedTimeRetryPolicy
+@see google::cloud::bigtable::DataLimitedErrorCountRetryPolicy
+@see google::cloud::bigtable_admin::BigtableTableAdminRetryPolicy
+@see google::cloud::bigtable_admin::BigtableInstanceAdminRetryPolicy
+
+@section bigtable-override-retry-backoff-policy Controlling the backoff algorithm
+
+The `*BackoffPolicyOption controls how long the client library will wait
```
Comment (dbolduc) [inline google/cloud/bigtable/doc/override-retry-policies.dox:60]: need closing backtick on `*BackoffPolicyOption`

**Review by dbolduc — APPROVED** [review]
Thanks.

File: google/cloud/bigtable/doc/override-retry-policies.dox [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+The `*IdempotencyPolicyOption` controls which requests are retryable, as some
+requests are never safe to retry.
+
+Only one built-in idempotency policy is provided by the library. The name
+matches the name of the client it is intended for. For example,
+`BigtableTableAdminClient` will use
+`BigtableTableAdminConnectionIdempotencyPolicy`.
+
+In the case of data operations, only mutations need to be consider.
```
[the remark this answers is not in the document]
Comment (coryan) [reply google/cloud/bigtable/doc/override-retry-policies.dox]: Fixed.

**Pull request merged**